src/array/array.jl

```diff
@@ -127,6 +127,23 @@ Base.similar(A::AbstractDimArray, ::Type{T}, i::Integer, I::Vararg{<:Integer}) w
     similar(A, T, (i, I...))
 Base.similar(A::AbstractDimArray, ::Type{T}, I::Tuple{Int,Vararg{Int}}) where T =
     similar(parent(A), T, I)
+# when the axes are DimUnitRanges we can return an `AbstractDimArray`
+function Base.similar(A::AbstractDimArray, T::Type, shape::Tuple{Dimensions.DimUnitRange,Vararg{Dimensions.DimUnitRange}})
+    data = similar(parent(A), T, map(parent, shape))
+    return rebuild(A; data=data, dims=deepcopy(dims(shape)), refdims=(), metadata=NoMetadata())
+end
+function Base.similar(
+    A::AbstractArray, T::Type, shape::Tuple{Dimensions.DimUnitRange,Vararg{Dimensions.DimUnitRange}}
+)
+    data = similar(parent(A), T, map(parent, shape))
+    return DimArray(data, deepcopy(dims(shape)))
```

Comment:
```suggestion
    C = dimconstructor(dims(shape))
    return C(data, deepcopy(dims(shape)))
```

We also have a trick for getting back other kinds of AbstractDimArray, as used in `rand` etc. It uses the maybe poorly named `dimconstructor` method:

https://github.com/rafaqz/DimensionalData.jl/blob/main/src/array/array.jl#L477-L481

It will return `DimArray` by default but allows packages to override that for `Dimension` or `LookupArray` that they define. So Rasters.jl will make a `Raster` here if any `LookupArray` is `Rasters.AbstractProjected`

https://github.com/rafaqz/Rasters.jl/blob/main/src/array.jl#L314